src/auth/strategies/oidc.strategy.ts

```diff
@@ -131,34 +145,57 @@ export class OidcStrategy extends PassportStrategy(Strategy, "oidc") {
     return returnUser;
   }
 
-  getUserPhoto(userinfo: UserinfoResponse) {
-    return userinfo.thumbnailPhoto
+  getUserPhoto(thumbnailPhoto: string) {
+    return thumbnailPhoto
       ? "data:image/jpeg;base64," +
-          Buffer.from(userinfo.thumbnailPhoto as string, "binary").toString(
-            "base64",
-          )
+          Buffer.from(thumbnailPhoto, "binary").toString("base64")
       : "no photo";
   }
 
-  parseUserInfo(userinfo: UserinfoResponse) {
+  parseUserInfo(userinfo: UserInfoResPonseWithGroups) {
     type OidcProfile = Profile & UserProfile;
     const profile = {} as OidcProfile;
 
+    const newUserInfoFields =
+      this.configService.get<IUserInfoMapping>("oidc.userInfoMapping") || {};
+
+    // To dynamically map user info fields based on environment variables,
+    // set mappings like OIDC_USERINFO_MAPPING_FIELD_USERNAME=family_name.
+    // This assigns userinfo.family_name to oidcUser.username.
+
+    const oidcUser: IUserInfoMapping = {
+      id: userinfo["sub"] || (userinfo["user_id"] as string) || "",
+      username: userinfo["sub"] || "",
+      displayName: userinfo["name"] || "",
+      familyName: userinfo["family_name"] || "",
+      email: userinfo["email"] || "",
+      thumbnailPhoto: (userinfo["thumbnailPhoto"] as string) || "",
```

Comment:
Do we get the photo from oidc?

Reply:
We can get photo from oidc, but I'm not sure if we are using it or not, this field was defined long ago

Reply:
The backward compatibility is done through the getUserPhoto function

getUserPhoto(thumbnailPhoto: string) {
  return thumbnailPhoto
    ? "data:image/jpeg;base64," +
        Buffer.from(thumbnailPhoto, "binary").toString("base64")
    : "no photo";
}